### PR TITLE
app-text/sigil: bump v0.9.16, improve ebuild

### PR DESCRIPTION
**Pull request by arthurzam**
1. Added more information to `metadata.xml`
2. In the commit of bump to `v0.9.16` it was mostly a copy of the previous `v0.9.13`, but changing dependency to `dev-qt/qtwebengine` and upping minimal Qt version to 5.12.
3. Added `USE=system-mathjax`. `Sigil` uses much smaller bundle of `mathjax` than `dev-libs/mathjax`. The difference is somewhere of between `70MB` and `1.2MB`. Please take note that the smaller bundle works only for `Sigil`'s use case.
4. Added `USE=plugins`. `Sigil` uses 6 Python packages as dependencies for enabling plugins support. Not everyone needs this support so make it optional.
5. Removed bundled `css_parser` and depend on `dev-python/css-parser` 

I think we shouldn't remove `0.9.13` for now from repository as it supports the deprecated `qtwebkit`, as some users might want to use it. But maybe we should stabilize it. Should I open stabilization bug for it on BugZilla? 

It is a self-maintained package, I don't know why this tag doesn't show?

Package-Manager: Portage-2.3.67, Repoman-2.3.14

### CODE REVIEW

File: app-text/sigil/sigil-0.9.16.ebuild [hunk 1]
```diff
@@ -4,6 +4,7 @@
 EAPI=7
 
 PYTHON_COMPAT=( python3_{5,6,7} )
+PYTHON_REQ_USE="tk"
```
Comment (a17r) [inline app-text/sigil/sigil-0.9.16.ebuild]: how does this change relate to `USE=plugins`? shouldn't it be in the version bump?

File: app-text/sigil/metadata.xml [hunk 2]
```diff
@@ -13,6 +13,7 @@
 		It is designed to edit books in ePub format.</longdescription>
 	<use>
 		<flag name="system-mathjax">Use the system-wide <pkg>dev-libs/mathjax</pkg> instead of bundled</flag>
+		<flag name="plugins">Installed needed Python packages for using Sigil plugins</flag>
```
Comment (a17r) [inline app-text/sigil/metadata.xml]: please keep them sorted

File: app-text/sigil/sigil-0.9.16.ebuild [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{5,6,7} )
+
+inherit cmake-utils python-single-r1 readme.gentoo-r1 xdg-utils
+
+MY_P="Sigil-${PV}"
+
+DESCRIPTION="Sigil is a multi-platform WYSIWYG ebook editor for ePub format"
```
Comment (a17r) [inline app-text/sigil/sigil-0.9.16.ebuild]: No use of `${PN}` in `DESCRIPTION` please. I'd recommend starting with `Multi-platform ...`

File: app-text/sigil/sigil-0.9.16.ebuild [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+HOMEPAGE="https://sigil-ebook.com/"
+SRC_URI="https://github.com/Sigil-Ebook/Sigil/archive/${PV}.tar.gz -> ${MY_P}.tar.gz"
```
Comment (a17r) [inline app-text/sigil/sigil-0.9.16.ebuild]: just use `${P}` for the tarball name here

File: app-text/sigil/sigil-0.9.16.ebuild [hunk 5]
```diff
@@ -0,0 +1,97 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{5,6,7} )
+
+inherit cmake-utils python-single-r1 readme.gentoo-r1 xdg-utils
+
+MY_P="Sigil-${PV}"
```
Comment (a17r) [inline app-text/sigil/sigil-0.9.16.ebuild]: better set `S` directly and drop this var.

File: app-text/sigil/sigil-0.9.16.ebuild [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	dev-python/html5lib[${PYTHON_USEDEP}]
+	dev-python/lxml[${PYTHON_USEDEP}]
+	dev-python/pillow[${PYTHON_USEDEP}]
+	dev-python/regex[${PYTHON_USEDEP}]
+	dev-python/six[${PYTHON_USEDEP}]
+	>=dev-qt/qtconcurrent-5.12:5
+	>=dev-qt/qtcore-5.12:5
+	>=dev-qt/qtgui-5.12:5
+	>=dev-qt/qtprintsupport-5.12:5
+	>=dev-qt/qtwebengine-5.12:5
```
Comment (a17r) [inline app-text/sigil/sigil-0.9.16.ebuild]: You'll likely have to depend on `qtwebengine[widgets]` here, did you check the buildsystem?

Comment (arthurzam) [reply app-text/sigil/sigil-0.9.16.ebuild]: Done.

File: app-text/sigil/sigil-0.9.16.ebuild (hunk 5 above)
Comment (arthurzam) [reply app-text/sigil/sigil-0.9.16.ebuild]: Done.

File: app-text/sigil/sigil-0.9.16.ebuild (hunk 4 above)
Comment (arthurzam) [reply app-text/sigil/sigil-0.9.16.ebuild]: Done.

File: app-text/sigil/sigil-0.9.16.ebuild (hunk 3 above)
Comment (arthurzam) [reply app-text/sigil/sigil-0.9.16.ebuild]: Done.

File: app-text/sigil/metadata.xml (hunk 2 above)
Comment (arthurzam) [reply app-text/sigil/metadata.xml]: Done

File: app-text/sigil/sigil-0.9.16.ebuild (hunk 1 above)
Comment (arthurzam) [reply app-text/sigil/sigil-0.9.16.ebuild]: I put it in the wrong commit as I found about this missing when changing `USE=plugins`.
Fixed.

**Comment by arthurzam** [comment]
Big thank you!

**Comment by a17r** [comment]
Thank you! ;)

**Pull request closed without merging**